spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -381,6 +381,100 @@
         subject.multi_receive(events)
       end
     end
+
+    context "unsupported actions" do
+      let(:options) { super().merge("index" => "logstash", "action" => "%{action_field}") }
+
+      context "with multiple valid actions with one trailing invalid action" do
+        let(:events) {[
+          LogStash::Event.new("action_field" => "index", "id" => 1, "message"=> "hello"),
+          LogStash::Event.new("action_field" => "index", "id" => 2, "message"=> "hi"),
+          LogStash::Event.new("action_field" => "index", "id" => 3, "message"=> "bye"),
+          LogStash::Event.new("action_field" => "unsupported_action", "id" => 4, "message"=> "world!")
+        ]}
+        it "rejects unsupported actions" do
+          event_result = subject.send(:safe_interpolation_map_events, events)
+          expect(event_result.successful_events.size).to be == 3
```

Comment:
@yaauie , I really liked the idea of having `rspec-collection_matchers` when unit tests check agains collections sizes. I understood more about use cases after taking a look at [this thread](https://github.com/rspec/rspec-expectations/issues/785).
However, in order to use `collection_matchers`, we need to onboard it with LS-core (add `gem "rspec-collection_matchers", :group => :development` in `Gemfile.template` file).

Let me know your thoughts if I can add it to plugin level dependency.

Reply:
I could go either way -- the failure scenario reads "clear enough" for these specs.

The normal way of adding the dependency would be to declare a development dependency in this project's gemspec, not in Logstash core.